### PR TITLE
[QUIC] Log msquic version and commit hash to event log and to test output.

### LINKED ISSUE #75596: QUIC and HTTP/3 tests should log which exact version of msquic they use
At least to the console -- that will allow us to better check in CI if the right version of msquic is installed (even in future).

### PR DESCRIPTION
**Pull request by ManickaP**
Fixes https://github.com/dotnet/runtime/issues/75596

### CODE REVIEW

**Review by CarnaViire — APPROVED** [review]
LGTM, thanks!

File: src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicApi.cs [hunk 1]
```diff
@@ -83,28 +89,58 @@ static MsQuicApi()
...
+            arraySize = 64;
+            paramSize = (uint)arraySize * sizeof(sbyte);
+            sbyte* libGitHash = stackalloc sbyte[arraySize];
+            status = apiTable->GetParam(null, QUIC_PARAM_GLOBAL_LIBRARY_GIT_HASH, &paramSize, libGitHash);
+            if (StatusFailed(status))
             {
                 if (NetEventSource.Log.IsEnabled())
                 {
-                    NetEventSource.Info(null, $"Incompatible MsQuic library version '{version}', expecting '{MsQuicVersion}'");
+                    NetEventSource.Error(null, $"Cannot retrieve {nameof(QUIC_PARAM_GLOBAL_LIBRARY_GIT_HASH)} from MsQuic library: '{status}'.");
```
Comment (CarnaViire) [inline src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicApi.cs:118]: Did you check it with both official and manually build package?

Comment (ManickaP) [reply src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicApi.cs:118]: CI is running with the official packages and I run locally with manually built msquic from the its main branch.

File: src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicApi.cs [hunk 2]
```diff
@@ -18,7 +18,7 @@ internal sealed unsafe partial class MsQuicApi
 {
     private static readonly Version MinWindowsVersion = new Version(10, 0, 20145, 1000);
 
-    private static readonly Version MsQuicVersion = new Version(2, 1);
+    private static readonly Version MinMsQuicVersion = new Version(2, 1);
```
Comment (stephentoub) [inline src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicApi.cs:21]: Nit:
s_minWindowsVersion
s_minMsQuicVersion

File: src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicApi.cs [hunk 3]
```diff
@@ -83,28 +89,58 @@ static MsQuicApi()
         try
         {
             // Check version
-            int arraySize = 4;
+            int arraySize;
+            uint paramSize;
+            int status;
+
+            arraySize = 4;
```
Comment (stephentoub) [inline src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicApi.cs:96]: It'd be better if these could be consts, e.g.
https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAMIR8AB14AbGFADKMgG68wMXAG4aNANr4IAEw7SQDWQGteYgDIQw2SbgCSXXhgC6m2gGYmpBkIYA3jQMIUze3LjYAGYwDLjAAJ4YsQCy5AAUAJSBwaF5kFy4GAy8PAwAglBQ2AmyvABesQC8DCga1Hl58UkwAFQMYhhQDC1F2GCmdpI2cYnJWpXVtQ0wru2docQA7ANDWnRruSEAvkcMZ8ThhdGx3ckMKaRZOR0bJWWLNXWNI63rb3c+rthqMMONJpJpmBZj0FlUvitDq8NttgfskXlTtRjkA==

Comment (ManickaP) [reply src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicApi.cs:96]: How much does this matter if this is called once per the whole application? Or is this more about setting up good examples/usage patterns?

File: src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicApi.cs (hunk 1 above)
Comment (ManickaP) [reply src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicApi.cs:118]: Note that this won't get hit unless the library is built without this property, which doesn't exists ATM. It's just checking for the status as we do with all msquic calls.

File: src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicApi.cs (hunk 2 above)
Comment (ManickaP) [reply src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicApi.cs]: Fixed

**Review by rzikm — APPROVED** [review]
LGTM :shipit:

**Pull request merged**